### ISSUE TITLE
panic with default config / setting inputsource that is not enabled in the OS

### ISSUE THREAD

**Issue by GTANAdam** [Serpentiel/betterglobekey#9]
**What happened?**

Paniced when pressed the globe key

**What did you expect to happen?**

Should just function as described

**What else do we need to know?**

Macbook Pro M1 (macos 12.6)

```
{"level":"info","ts":1667535604.2773812,"caller":"cmd/root.go:83","msg":"betterglobekey has been started"}
{"level":"info","ts":1667535604.277499,"caller":"eventlistener/eventlistener.go:16","msg":"eventlistener has been initialized"}
{"level":"info","ts":1667535604.277576,"caller":"eventlistener/eventlistener.go:20","msg":"eventhandler has been initialized"}
{"level":"info","ts":1667535606.651946,"caller":"eventhandler/eventhandler.go:115","msg":"globe key pressed"}
SIGTRAP: trace trap
PC=0x1b934d44c m=5 sigcode=0
signal arrived during cgo execution

goroutine 19 [syscall]:
runtime.cgocall(0x104f6ee70, 0x14000147c88)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/cgocall.go:158 +0x54 fp=0x14000147c50 sp=0x14000147c10 pc=0x104c742f4
github.com/Serpentiel/betterglobekey/pkg/inputsource._Cfunc_CFRelease(0x0)
	_cgo_gotypes.go:242 +0x34 fp=0x14000147c80 sp=0x14000147c50 pc=0x104d4b474
github.com/Serpentiel/betterglobekey/pkg/inputsource.Select.func8.1()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/pkg/inputsource/inputsource.go:87 +0x24 fp=0x14000147ca0 sp=0x14000147c80 pc=0x104d4c584
github.com/Serpentiel/betterglobekey/pkg/inputsource.Select({0x140001323a8, 0x16})
	/private/tmp/betterglobekey-20221104-97443-1uyht63/pkg/inputsource/inputsource.go:92 +0x108 fp=0x14000147d10 sp=0x14000147ca0 pc=0x104d4c478
github.com/Serpentiel/betterglobekey/internal/pkg/eventhandler.(*fnKeyHandler).setInputSource(0x1400012c480, {0x140001323a8, 0x16})
	/private/tmp/betterglobekey-20221104-97443-1uyht63/internal/pkg/eventhandler/eventhandler.go:86 +0x70 fp=0x14000147dc0 sp=0x14000147d10 pc=0x104f6b9c0
github.com/Serpentiel/betterglobekey/internal/pkg/eventhandler.(*fnKeyHandler).KeyUp.func1()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/internal/pkg/eventhandler/eventhandler.go:117 +0x24c fp=0x14000147e80 sp=0x14000147dc0 pc=0x104f6bd7c
github.com/Serpentiel/betterglobekey/internal/pkg/eventlistener.Run(0x1400012a0c8)
	/private/tmp/betterglobekey-20221104-97443-1uyht63/internal/pkg/eventlistener/eventlistener.go:29 +0x250 fp=0x14000147fb0 sp=0x14000147e80 pc=0x104f6e3c0
github.com/Serpentiel/betterglobekey/cmd.runRootCmd.func1()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/cmd/root.go:85 +0x28 fp=0x14000147fd0 sp=0x14000147fb0 pc=0x104f6ecc8
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000147fd0 sp=0x14000147fd0 pc=0x104cd4824
created by github.com/Serpentiel/betterglobekey/cmd.runRootCmd
	/private/tmp/betterglobekey-20221104-97443-1uyht63/cmd/root.go:85 +0x140

goroutine 1 [select (no cases)]:
runtime.gopark(0x104cb0360?, 0x1400011b7b0?, 0xa0?, 0x21?, 0x104f6e930?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x1400015db50 sp=0x1400015db30 pc=0x104ca79f4
runtime.block()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/select.go:104 +0x30 fp=0x1400015db80 sp=0x1400015db50 pc=0x104cb7120
github.com/Serpentiel/betterglobekey/cmd.runRootCmd(0x1052efc80?, {0x104f76114?, 0x0?, 0x0?})
	/private/tmp/betterglobekey-20221104-97443-1uyht63/cmd/root.go:87 +0x144 fp=0x1400015dd30 sp=0x1400015db80 pc=0x104f6e934
github.com/spf13/cobra.(*Command).execute(0x1052efc80, {0x1400011a1e0, 0x0, 0x0})
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x4b8 fp=0x1400015de10 sp=0x1400015dd30 pc=0x104dee728
github.com/spf13/cobra.(*Command).ExecuteC(0x1052efc80)
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x354 fp=0x1400015ded0 sp=0x1400015de10 pc=0x104deed24
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/Serpentiel/betterglobekey/cmd.Execute()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/cmd/root.go:34 +0x17c fp=0x1400015df60 sp=0x1400015ded0 pc=0x104f6e69c
main.main()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/main.go:8 +0x1c fp=0x1400015df70 sp=0x1400015df60 pc=0x104f6ed1c
runtime.main()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:250 +0x24c fp=0x1400015dfd0 sp=0x1400015df70 pc=0x104ca762c
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x1400015dfd0 sp=0x1400015dfd0 pc=0x104cd4824

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000048fa0 sp=0x14000048f80 pc=0x104ca79f4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:369
runtime.forcegchelper()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:302 +0xac fp=0x14000048fd0 sp=0x14000048fa0 pc=0x104ca788c
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000048fd0 sp=0x14000048fd0 pc=0x104cd4824
created by runtime.init.6
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:290 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049770 sp=0x14000049750 pc=0x104ca79f4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:369
runtime.bgsweep(0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgcsweep.go:278 +0xa4 fp=0x140000497b0 sp=0x14000049770 pc=0x104c94b34
runtime.gcenable.func1()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgc.go:178 +0x28 fp=0x140000497d0 sp=0x140000497b0 pc=0x104c890f8
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000497d0 sp=0x140000497d0 pc=0x104cd4824
created by runtime.gcenable
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgc.go:178 +0x74

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x14000070000?, 0x10501c180?, 0x1?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049f50 sp=0x14000049f30 pc=0x104ca79f4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:369
runtime.(*scavengerState).park(0x1052f8480)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgcscavenge.go:389 +0x5c fp=0x14000049f80 sp=0x14000049f50 pc=0x104c92aec
runtime.bgscavenge(0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgcscavenge.go:617 +0x44 fp=0x14000049fb0 sp=0x14000049f80 pc=0x104c93094
runtime.gcenable.func2()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgc.go:179 +0x28 fp=0x14000049fd0 sp=0x14000049fb0 pc=0x104c89098
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000049fd0 sp=0x14000049fd0 pc=0x104cd4824
created by runtime.gcenable
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mgc.go:179 +0xb8

goroutine 18 [finalizer wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000044580 sp=0x14000044560 pc=0x104ca79f4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:369
runtime.runfinq()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mfinal.go:180 +0x120 fp=0x140000447d0 sp=0x14000044580 pc=0x104c88320
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000447d0 sp=0x140000447d0 pc=0x104cd4824
created by runtime.createfing
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/mfinal.go:157 +0x84

goroutine 20 [syscall]:
runtime.cgocall(0x104f74580, 0x140000457c8)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/cgocall.go:158 +0x54 fp=0x14000045790 sp=0x14000045750 pc=0x104c742f4
github.com/robotn/gohook._Cfunc_start_ev()
	_cgo_gotypes.go:135 +0x34 fp=0x140000457c0 sp=0x14000045790 pc=0x104f6ce64
github.com/robotn/gohook.Start.func2()
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/robotn/gohook@v0.40.0/hook.go:225 +0x24 fp=0x140000457d0 sp=0x140000457c0 pc=0x104f6df74
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000457d0 sp=0x140000457d0 pc=0x104cd4824
created by github.com/robotn/gohook.Start
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/robotn/gohook@v0.40.0/hook.go:225 +0x64

goroutine 21 [sleep]:
runtime.gopark(0x8b976fd5518f?, 0x104f6d930?, 0x7c?, 0x45?, 0x14000143fb8?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000143f70 sp=0x14000143f50 pc=0x104ca79f4
time.Sleep(0x2faf080)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/time.go:195 +0x118 fp=0x14000143fb0 sp=0x14000143f70 pc=0x104cd1378
github.com/robotn/gohook.Start.func1()
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/robotn/gohook@v0.40.0/hook.go:235 +0x2c fp=0x14000143fd0 sp=0x14000143fb0 pc=0x104f6d93c
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000143fd0 sp=0x14000143fd0 pc=0x104cd4824
created by github.com/robotn/gohook.Start
	/Users/adam/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/robotn/gohook@v0.40.0/hook.go:228 +0x80

goroutine 5 [chan receive]:
runtime.gopark(0x14000103860?, 0x300000002?, 0x28?, 0x5f?, 0x104cd2198?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000045ee0 sp=0x14000045ec0 pc=0x104ca79f4
runtime.chanrecv(0x14000072060, 0x0, 0x1)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/chan.go:583 +0x454 fp=0x14000045f70 sp=0x14000045ee0 pc=0x104c77084
runtime.chanrecv1(0x0?, 0x104cd4824?)
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/chan.go:442 +0x14 fp=0x14000045fa0 sp=0x14000045f70 pc=0x104c76bf4
github.com/Serpentiel/betterglobekey/internal/pkg/eventhandler.(*fnKeyHandler).KeyUp.func1.1()
	/private/tmp/betterglobekey-20221104-97443-1uyht63/internal/pkg/eventhandler/eventhandler.go:102 +0x34 fp=0x14000045fd0 sp=0x14000045fa0 pc=0x104f6bf44
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.3/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000045fd0 sp=0x14000045fd0 pc=0x104cd4824
created by github.com/Serpentiel/betterglobekey/internal/pkg/eventhandler.(*fnKeyHandler).KeyUp.func1
	/private/tmp/betterglobekey-20221104-97443-1uyht63/internal/pkg/eventhandler/eventhandler.go:101 +0xac

r0      0x0
r1      0x104f6ee70
r2      0x14000147c00
r3      0x14000003520
r4      0x400
r5      0x140001035b0
r6      0x105327a37
r7      0x1
r8      0x1b9576e91
r9      0x20f357000
r10     0x100000211b8a96d
r11     0x200000211b8a96d
r12     0x1c2112392
r13     0x16b3bef00
r14     0x1c2112390
r15     0x1c2112390
r16     0x1b91801c0
r17     0x212894108
r18     0x0
r19     0x14
r20     0x20
r21     0x104f6bd30
r22     0x104f6e3c0
r23     0x104fe0c3c
r24     0xffffffffffffffff
r25     0x104fb5c20
r26     0x1400001c4a0
r27     0x858
r28     0x14000003520
r29     0x16b3beef0
lr      0x1b91802cc
sp      0x16b3beef0
pc      0x1b934d44c
fault   0x1b934d44c
```

**Comment by Serpentiel** [comment]
hey 👋, thank you for reporting this!

could you please share the config file located at `~/.betterglobekey.yaml`?

**Comment by Serpentiel** [comment]
hey again, @GTANAdam!

it would be helpful if you could provide me with the file that you use :)

**Comment by GTANAdam** [comment]
Greetings
```
# Double press
double_press:
  # Maximum delay between two presses of the Globe key for them to be considered as a double press
  maximum_delay: 250

# Input sources
input_sources:
  # Primary input sources
  # This is used when the Globe key has been pressed once
  primary:
    - com.apple.keylayout.US
    - com.apple.keylayout.Russian

  # Additional input sources
  # This is used when the Globe key has been double pressed
  additional:
    - com.apple.keylayout.Finnish
```

**Comment by Serpentiel** [comment]
thank you, I will add extra sanity checks to address the problem, but I would like to suggest you to edit the config to make the tool work, please do in the Terminal:

```bash
❯ betterglobekey list
com.apple.inputmethod.Kotoeri.RomajiTyping.Japanese
com.apple.keylayout.Finnish
com.apple.keylayout.Russian
com.apple.keylayout.US
```

and fill the config file `~/.betterglobekey.yaml` with the input sources from the outputted list

**Comment by GTANAdam** [comment]
That fixes it. I would suggest auto pre-filling this on configuration file creation.
I would also suggest allowing the empty `additional` inputs, considering that `additional: []` results in the following when double pressing:
```
panic: runtime error: index out of range [0] with length 0
```

**Comment by Serpentiel** [comment]
thank you for contributing to the project!

this issue will be closed once the PR is merged :)

**Comment by Serpentiel** [comment]
@GTANAdam this is now resolved as of v2.1.0
